cpp/src/arrow/compute/api_scalar.h

```diff
@@ -117,6 +119,20 @@ class ARROW_EXPORT RoundTemporalOptions : public FunctionOptions {
   CalendarUnit unit;
   /// What day does the week start with (Monday=true, Sunday=false)
   bool week_starts_monday;
+  /// Enable this flag to return a rounded value that is strictly greater than the input.
+  /// For example: ceiling 1970-01-01T00:00:00 to 3 hours would yield 1970-01-01T03:00:00
+  /// if set to true and 1970-01-01T00:00:00 if set to false.
+  /// This applies for ceiling only.
+  bool ceil_is_strictly_greater;
+  /// By default time is rounded to a multiple of units since 1970-01-01T00:00:00.
+  /// By setting multiple_since_greater_unit to true, time will be rounded to a number
+  /// of units since the last greater calendar unit.
+  /// For example: rounding to a multiple of days since the beginning of the month or
+  /// to hours since the beginning of the day.
+  /// Exceptions: week and quarter are not used as greater units, therefore days will
+  /// will be rounded to the beginning of the month not week. Greater unit of week
+  /// is year.
+  bool multiple_since_greater_unit;
```

Comment:
I'm not fond of `multiple_since_greater_unit` but I cannot think of a vastly better name and the explanation is clear, so we can probably keep it as is.

Reply:
Perhaps `calendar_based`? cc @jorisvandenbossche

Reply:
FWIW I like the calendar association but  `multiple_since_greater_unit` is more idiomatic then other suggestions so far. Since this is likely to be an internal parameter I'm ok with either option.

Reply:
@pitrou is this the last point that needs to be addressed here?

Reply:
Nothing needs to be addressed a priori, just waiting for @jorisvandenbossche 's opinion.

Reply:
So in the beginning of the PR we had `calendar_based_origin`, but then we changed it triggered by a comment of mine: https://github.com/apache/arrow/pull/12657#discussion_r866026624

Now I better understand the keyword (and the explanation in the docs is better), I think `calendar_based_origin` is actually maybe quite OK. I like that "origin" is in it.